src/openforms/registrations/contrib/objects_api/plugin.py

```diff
@@ -161,7 +160,29 @@ def register_submission(
                 get_drc=get_drc,
                 language=language_code_2b,
             )
-            record_data["csv_url"] = submission_csv_document["url"]
+            csv_url = submission_csv_document["url"]
+
+        base_render_context = {
+            "_submission": submission,
+            **get_variables_for_context(submission),
+            "type": options["productaanvraag_type"],
+            "submission": {
+                "uuid": submission.uuid,
+                "language_code": submission.language_code,
+                "attachments": attachments,
+                "pdf_url": document["url"],
+                "csv_url": csv_url,
+            },
+        }
+
+        record_data = {
+            "data": render_from_string(
+                templates,
+                context={**base_render_context},
+                disable_autoescape=True,
+                backend=openforms_backend,
+            ),
+        }
```

Comment:
There should be no need to nest the rendered json in the "data" because on line 194 you already have the key data (but I guess this is still work in progress)